src/utils/src/Thread.c

```diff
@@ -159,14 +176,17 @@ PUBLIC_API STATUS defaultCreateThread(PTID pThreadId, startRoutine start, PVOID
 
     CHK(pThreadId != NULL, STATUS_NULL_ARG);
 
-#ifdef CONSTRAINED_DEVICE
     pthread_attr_t attr;
-    pAttr = &attr;
-    result = pthread_attr_init(pAttr);
-    CHK_ERR(result == 0, STATUS_THREAD_ATTR_INIT_FAILED, "pthread_attr_init failed with %d", result);
-    result = pthread_attr_setstacksize(&attr, THREAD_STACK_SIZE_ON_CONSTRAINED_DEVICE);
-    CHK_ERR(result == 0, STATUS_THREAD_ATTR_SET_STACK_SIZE_FAILED, "pthread_attr_setstacksize failed with %d", result);
-#endif
+    if (stackSize != 0) {
+        if (stackSize < (SIZE_T) PTHREAD_MIN_STACK_SIZE) {
+            stackSize = (SIZE_T) PTHREAD_MIN_STACK_SIZE;
```

Comment:
I would recommend logging or exit with some exception if given a value lower than the min.

Reply:
There seems to be a difference of opinion on how to tackle this error. Either let the pthread call fail and log it, or correct it to the allowed minimum.

There are valid points for both, I'll stick with the current implementation.